ui/StatusQ/src/StatusQ/Popups/StatusMenuItemDelegate.qml

```diff
@@ -7,187 +7,164 @@ import StatusQ.Components 0.1
 import StatusQ.Popups 0.1
 
 MenuItem {
-    id: statusPopupMenuItem
+    id: root
     implicitWidth: parent ? parent.width : 0
-    implicitHeight: action.enabled ? 34 : 0
-    objectName: action.objectName
-
-    property int subMenuIndex
-    property var statusPopupMenu: null
-
-    Component.onCompleted: {
-        if (!!subMenu) {
-            subMenuIndex = statusPopupMenu.menuItemCount
-            statusPopupMenu.menuItemCount += 1
+    implicitHeight: menu.hideDisabledItems && !enabled ? 0 : 38
+    objectName: action ? action.objectName : "StatusMenuItemDelegate"
+
+    spacing: 4
+    horizontalPadding: 8
+
+    readonly property string logObjectName: "StatusMenuItemDelegate [%1, %2]".arg(this).arg(text)
+
+    QtObject {
+        id: d
+
+        readonly property bool isSubMenu: !!root.subMenu
+        readonly property bool isStatusSubMenu: isSubMenu && (root.subMenu instanceof StatusPopupMenu)
+        readonly property bool subMenuOpened: isSubMenu && root.subMenu.opened
+        readonly property bool hasAction: !!root.action
+        readonly property bool isStatusAction: d.hasAction && (root.action instanceof StatusMenuItem)
+        readonly property bool isDangerIcon: d.isStatusAction && root.action.type === StatusMenuItem.Type.Danger
```

Comment:
```suggestion
        readonly property bool isStatusDangerAction: d.isStatusAction && root.action.type === StatusMenuItem.Type.Danger
```

Reply:
Oh yeah, QtC offers a great variety of QML refactoring tools :smiling_imp: not...